Misspelled new.target causes assertion failure

**TypeScript Version:**  2.2.1

**Code**

```ts
new.tagret
```

**Expected behavior:**

"error TS17012: 'tagret' is not a valid meta-property for keyword 'new'. Did you mean 'target'?"

**Actual behavior:**

```
Error: Debug Failure. False expression: Unrecognized meta-property.
    at Object.assert (/home/jeffrey/TypeScript-2.2.1/lib/tsc.js:1888:23)
    at checkMetaProperty (/home/jeffrey/TypeScript-2.2.1/lib/tsc.js:31127:22)
    at checkExpressionWorker (/home/jeffrey/TypeScript-2.2.1/lib/tsc.js:32267:28)
    at checkExpression (/home/jeffrey/TypeScript-2.2.1/lib/tsc.js:32205:42)
    at checkExpressionStatement (/home/jeffrey/TypeScript-2.2.1/lib/tsc.js:33857:13)
    at checkSourceElement (/home/jeffrey/TypeScript-2.2.1/lib/tsc.js:35324:28)
    at Object.forEach (/home/jeffrey/TypeScript-2.2.1/lib/tsc.js:278:30)
    at checkSourceFileWorker (/home/jeffrey/TypeScript-2.2.1/lib/tsc.js:35426:20)
    at checkSourceFile (/home/jeffrey/TypeScript-2.2.1/lib/tsc.js:35411:13)
    at getDiagnosticsWorker (/home/jeffrey/TypeScript-2.2.1/lib/tsc.js:35464:17)
```

--- a/src/compiler/checker.ts
+++ b/src/compiler/checker.ts
@@ -14804,7 +14804,6 @@ namespace ts {
 
         function checkMetaProperty(node: MetaProperty) {
             checkGrammarMetaProperty(node);
-            Debug.assert(node.keywordToken === SyntaxKind.NewKeyword && node.name.text === "target", "Unrecognized meta-property.");
             const container = getNewTargetContainer(node);
             if (!container) {
                 error(node, Diagnostics.Meta_property_0_is_only_allowed_in_the_body_of_a_function_declaration_function_expression_or_constructor, "new.target");
@@ -23021,7 +23020,7 @@ namespace ts {
         function checkGrammarMetaProperty(node: MetaProperty) {
             if (node.keywordToken === SyntaxKind.NewKeyword) {
                 if (node.name.text !== "target") {
-                    return grammarErrorOnNode(node.name, Diagnostics._0_is_not_a_valid_meta_property_for_keyword_1_Did_you_mean_0, node.name.text, tokenToString(node.keywordToken), "target");
+                    return grammarErrorOnNode(node.name, Diagnostics._0_is_not_a_valid_meta_property_for_keyword_1_Did_you_mean_2, node.name.text, tokenToString(node.keywordToken), "target");
                 }
             }
         }

--- a/src/compiler/diagnosticMessages.json
+++ b/src/compiler/diagnosticMessages.json
@@ -3269,7 +3269,7 @@
         "category": "Error",
         "code": 17011
     },
-    "'{0}' is not a valid meta-property for keyword '{1}'. Did you mean '{0}'?": {
+    "'{0}' is not a valid meta-property for keyword '{1}'. Did you mean '{2}'?": {
         "category": "Error",
         "code": 17012
     },

--- a/tests/baselines/reference/misspelledNewMetaProperty.errors.txt
+++ b/tests/baselines/reference/misspelledNewMetaProperty.errors.txt
@@ -1,0 +1,7 @@
+tests/cases/compiler/misspelledNewMetaProperty.ts(1,20): error TS17012: 'targ' is not a valid meta-property for keyword 'new'. Did you mean 'target'?
+
+
+==== tests/cases/compiler/misspelledNewMetaProperty.ts (1 errors) ====
+    function foo(){new.targ}
+                       ~~~~
+!!! error TS17012: 'targ' is not a valid meta-property for keyword 'new'. Did you mean 'target'?

--- a/tests/baselines/reference/misspelledNewMetaProperty.js
+++ b/tests/baselines/reference/misspelledNewMetaProperty.js
@@ -1,0 +1,5 @@
+//// [misspelledNewMetaProperty.ts]
+function foo(){new.targ}
+
+//// [misspelledNewMetaProperty.js]
+function foo() { new.targ; }

--- a/tests/cases/compiler/misspelledNewMetaProperty.ts
+++ b/tests/cases/compiler/misspelledNewMetaProperty.ts
@@ -1,0 +1,1 @@
+﻿function foo(){new.targ}

